### PR TITLE
bpo-20104: Remove posix_spawn from 3.7

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue20104

### CODE REVIEW

**Comment by pablogsal** [comment]
CC: @gpshead

**Review by gpshead — COMMENTED** [review]
this looks straight forward.

I suggest dropping the changes to aclocal.m4, configure.ac, configure, pyconfig.h.in that remove the check for the API.  It is fine for us to leave the autoconf script in 3.7 testing for the posix_spawn function.

**Comment by gpshead** [comment]
okay, i've undone the configure changes and done the news entrys.  just waiting for the various CI runs to complete (overloaded by the sprints) before merging.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**
